### ISSUE TITLE
Add id prop

### ISSUE THREAD

**Issue by sawyerclick** [janosh/svelte-bricks#10]
It'd be helpful to be able to pass a string or function to use an already-existing id column in an array of objects that may be under a different name. The easiest to implement would be a string to use for the identification column.

Something like the following:

```svelte
<Masonry items={items} id={(item) => item.columnName}/>
<Masonry items={items} id={'columnName'}/>
```

**Comment by janosh** [comment]
Is your data immutable? If not, couldn't you do this?

```svelte
<Masonry items={items.map((itm) => ({ ...itm, id: itm.columnName }))} />
```

**Comment by janosh** [comment]
Actually never mind mutability. This leaves original data unchanged so the only consideration would be memory but then this component is probably a bad fit anyway.

**Comment by sawyerclick** [comment]
My data isn't immutable, and I could technically use a `items.map()` approach. It feels like an unnecessary step though if a unique identifier can already be found in the object. There is a bottleneck if you have enough items and need to re-iterate, but I see this mostly as an enhancement/convenience feature.

**Comment by sawyerclick** [comment]
I think the getId function could possibly be a prop to allow custom identifiers across all types (which is probably overkill but would likely fit someone's use-case)

```svelte
  export let getId = (item: Item): string | number | undefined => {
    if (typeof item === `string`) return item
    if (typeof item === `number`) return item
    return item.id
  }
```

```svelte
<Masonry items={items} getId={(item) => (item.columnName)}/>
```

**Comment by janosh** [comment]
I considered making that change but it wouldn't support specifying a custom string for the id key. I guess we can add an additional prop `idKey` to change that behavior in `getId`.

**Comment by janosh** [comment]
Released in v0.1.7.
